src/components/Form/Form.tsx

```diff
@@ -19,6 +24,14 @@ export const Form = ({ name }: FormProps) => {
 
   function handleSubmit(e: React.FormEvent<HTMLFormElement>) {
     e.preventDefault();
+    try {
+      handleStorage(login, password);
+      navigate("/");
```

Comment:
Форма не должна сама принимать никаких решений по навигации, как мне кажется. Она должна лишь вызвать функцию, которую её передали в пропсы. Предлагаю сделать так. navigate("/"); перенести в сайн ин и сайн ап после диспатча.
```suggestion
      onSubmit(login, password);
```